tests/util.py

```diff
@@ -0,0 +1,32 @@
+import os
+import shutil
+import codecs
+from contextlib import contextmanager
+
+from sphinx.application import Sphinx
+
+
+@contextmanager
+def sphinx_build(test_dir, builder='html'):
+    os.chdir('tests/{0}'.format(test_dir))
+    try:
+        app = Sphinx(
+            srcdir='.',
+            confdir='.',
+            outdir='_build/{0}'.format(builder),
+            doctreedir='_build/.doctrees',
+            buildername=builder,
+        )
+        app.build(force_all=True)
+        yield app
+    finally:
+        shutil.rmtree('_build')
+        os.chdir('../..')
```

Comment:
I haven't had time to play around with `sphinx-testing`, i've only known of its existence. It would probably make sense to become acquainted with it and standardize on it. Sphinx core, and to a subset sphinx-testing, have some better patterns for testing doctress/etc.